luavm/src/lib/lua_values/lua_obj.rs

```diff
@@ -32,13 +43,25 @@ impl LuaObj for LuaInt {
         Ok(self.v)
     }
 
+    fn is_number(&self) -> bool {
```

Comment:
How can `is_number` be true but `is_float` below be false?

Reply:
Hah, I guess the name `is_float` is quite misleading. If you have an arithmetic expression, and either of the two operands is a float or a string, both arguments are converted to floats. `is_float` is used to check whether an operand is a float or a string. I guess I should rename this into something more descriptive.

Reply:
Yes, a different name would be good.